query_functions/src/selectors.rs

```diff
@@ -834,7 +985,23 @@ mod test {
                 Arc::clone(&schema),
                 vec![
                     Arc::new(Float64Array::from(vec![Some(2.0), Some(4.0), None])),
+                    Arc::new(Float64Array::from(vec![
+                        Some(f64::NAN),
+                        Some(f64::INFINITY),
+                        Some(f64::NEG_INFINITY),
+                    ])),
+                    Arc::new(Float64Array::from(vec![
+                        Some(f64::NEG_INFINITY),
+                        Some(f64::NEG_INFINITY),
+                        Some(f64::NEG_INFINITY),
+                    ])),
+                    Arc::new(Float64Array::from(vec![
+                        Some(f64::NAN),
+                        Some(f64::NAN),
+                        Some(f64::NAN),
```

Comment:
I recommend putting some different values as  the first/last values here. Something like 

```suggestion
                    Arc::new(Float64Array::from(vec![
                        Some(1.0),
                        Some(f64::NEG_INFINITY),
                        Some(f64::NEG_INFINITY),
                    ])),
                    Arc::new(Float64Array::from(vec![
                        Some(2.0,
                        Some(f64::NAN),
                        Some(f64::NAN),
```

To ensure that first/last selectors are really picking the right thing.

Or maybe it would be better as  a different twar case that is needed 🤔